src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -79,6 +81,17 @@ internal InfiniteScrollList(Lazy<JoinableTaskFactory> joinableTaskFactory)
 
             DataContext = Items;
             CheckBoxesEnabled = false;
+
+            _loadingStatusIndicator.PropertyChanged += LoadingStatusIndicator_PropertyChanged;
+        }
+
+        private void LoadingStatusIndicator_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(LoadingStatusIndicator.Status))
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
This check should not happen in conditional checks. 
It should always be at the top of the method. 

The state should not be affecting the affinity expectations.

Reply:
Moved to the top of the method.